agent/lib/kontena/websocket_client.rb

```diff
@@ -177,5 +182,28 @@ def verify_connection
     rescue => exc
       error exc.message
     end
+
+    # Abort the connection, closing the websocket, with a timeout
+    def close
+      return if @close_timer
+
+      # stop sending messages, queue them up until reconnected
+      Celluloid::Notifications.publish('websocket:disconnect', nil)
+
+      # send close frame; this will get stuck if the server is not replying
+      ws.close(1000)
```

Comment:
Oh, yes.. I did not mean to call it twice. But now I just realized that close does not block ever.